bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/factory/BasicModuleHandlerFactory.java

```diff
@@ -231,10 +232,18 @@ protected ModuleHandler internalCreate(Module module, String ruleUID) {
             CompareConditionHandler compareConditionHandler = handler != null
                     && handler instanceof CompareConditionHandler ? (CompareConditionHandler) handler : null;
             if (compareConditionHandler == null) {
-                compareConditionHandler = new CompareConditionHandler((Condition)module);
-                handlers.put(ruleUID+module.getId(), compareConditionHandler);
+                compareConditionHandler = new CompareConditionHandler((Condition) module);
+                handlers.put(ruleUID + module.getId(), compareConditionHandler);
             }
             return compareConditionHandler;
+        } else if (TimerTriggerHandler.MODULE_TYPE_ID.equals(moduleTypeUID) && module instanceof Trigger) {
+            TimerTriggerHandler timerTriggerHandler = handler != null && handler instanceof TimerTriggerHandler
+                    ? (TimerTriggerHandler) handler : null;
+            if (timerTriggerHandler == null) {
+                timerTriggerHandler = new TimerTriggerHandler((Trigger) module);
```

Comment:
if the org.quartz import is not satisfied (as it is optional), this will result in a NCDFE, which needs to be handled here. Remember that the module type must only be registered if Quartz is available.